### DISCUSSION TITLE
Is it possible to return the FLUID job id from `flux-submit` as decimal?

### DISCUSSION THREAD

**Discussion by dblitt** [flux-framework/flux-core#6429] [Q&A] [answered]
I am currently working on an integration for Flux into [`cctools/batch_queue`](https://cctools.readthedocs.io/en/stable/api/html/batch__queue_8h.html) so we can easily submit generic jobs into a Flux instance from this shared API. The current plan is basically to call `popen("flux submit ...")` (so as to be as portable as possible) and parse the output to obtain the jobid, however it seems that flux-submit always returns the jobid in F58 format (possibly in unicode), and I would rather not have to parse this if not necessary, because our internal format keeps the jobid as a decimal.

**Comment by grondo** [comment, marked as answer]
You could popen the following command line:
```console
$ flux submit -n4 hostname | flux job id --to=dec
353630158848
```

Note that the integer returned here is 64 bit unsigned where it looks like cctools uses `int64_t` from a peek at that link. Just something to keep in mind.

Alternately,  the Flux API does export `flux_job_id_parse(3)` which can parse a jobid in any format into `flux_jobid_t` (which is a typedef of `uint64_t`).
